test_scripts/RC/commonRC.lua

```diff
@@ -0,0 +1,740 @@
+          :Times(AtLeast(1)) -- issue with SDL --> notification is sent twice
+          self["mobileSession" .. id]:ExpectNotification("OnPermissionsChange")
+        end)
+    end)
+end
+
+function commonRC.unregisterApp(pAppId, self)
+  local mobSession = commonRC.getMobileSession(self, pAppId)
+  local hmiAppId = commonRC.getHMIAppId(pAppId)
+  local cid = mobSession:SendRPC("UnregisterAppInterface",{})
+  EXPECT_HMINOTIFICATION("BasicCommunication.OnAppUnregistered", { appID = hmiAppId, unexpectedDisconnect = false })
+  mobSession:ExpectResponse(cid, { success = true, resultCode = "SUCCESS"})
+end
+
+function commonRC.activate_app(pAppId, self)
+  self, pAppId = commonRC.getSelfAndParams(pAppId, self)
+  if not pAppId then pAppId = 1 end
+  local pHMIAppId = hmiAppIds[config["application" .. pAppId].registerAppInterfaceParams.appID]
+  local mobSession = commonRC.getMobileSession(self, pAppId)
+  local requestId = self.hmiConnection:SendRequest("SDL.ActivateApp", { appID = pHMIAppId })
+  EXPECT_HMIRESPONSE(requestId)
+  mobSession:ExpectNotification("OnHMIStatus", { hmiLevel = "FULL", audioStreamingState = "AUDIBLE", systemContext = "MAIN" })
+  commonTestCases:DelayedExp(commonRC.minTimeout)
+end
+
+function commonRC.postconditions()
+  StopSDL()
+end
+
+function commonRC.getSelfAndParams(...)
+  local out = { }
+  local selfIdx = nil
+  for i,v in pairs({...}) do
+    if type(v) == "table" and v.isTest then
+      table.insert(out, v)
+      selfIdx = i
+      break
+    end
+  end
+  local idx = 2
+  for i = 1, table.maxn({...}) do
+    if i ~= selfIdx then
+      out[idx] = ({...})[i]
+      idx = idx + 1
+    end
+  end
+  return table.unpack(out, 1, table.maxn(out))
+end
+
+function commonRC.getModuleControlData(module_type)
+  local out = { moduleType = module_type }
+  if module_type == "CLIMATE" then
+    out.climateControlData = {
+      fanSpeed = 50,
+      currentTemperature = {
+        unit = "FAHRENHEIT",
+        value = 20.1
+      },
+      desiredTemperature = {
+        unit = "CELSIUS",
+        value = 10.5
+      },
+      acEnable = true,
+      circulateAirEnable = true,
+      autoModeEnable = true,
+      defrostZone = "FRONT",
+      dualModeEnable = true,
+      acMaxEnable = true,
+      ventilationMode = "BOTH"
+    }
+  elseif module_type == "RADIO" then
+    out.radioControlData = {
+      frequencyInteger = 1,
+      frequencyFraction = 2,
+      band = "AM",
+      rdsData = {
+        PS = "ps",
+        RT = "rt",
+        CT = "123456789012345678901234",
+        PI = "pi",
+        PTY = 1,
+        TP = false,
+        TA = true,
+        REG = "US"
+      },
+      availableHDs = 1,
+      hdChannel = 1,
+      signalStrength = 5,
+      signalChangeThreshold = 10,
+      radioEnable = true,
+      state = "ACQUIRING"
+    }
+  end
+  return out
+end
+
+function commonRC.getAnotherModuleControlData(module_type)
+  local out = { moduleType = module_type }
```

Comment:
Looks like this is just slightly different sample data for the module. Probably could be named better.